test/extensions/filters/http/local_ratelimit/local_ratelimit_integration_test.cc

```diff
@@ -315,5 +431,44 @@ TEST_P(LocalRateLimitFilterIntegrationTest, BasicTestPerRouteAndRds) {
   cleanUpXdsConnection();
 }
 
+#ifdef NDEBUG
```

Comment:
Sorry for the late observation, but IMHO this is a red-flag when it comes to tests (especially integration tests). Which assertion in the singleton manager is triggered?

Specifically I would like to see a validation of the MAIN_THREAD assertions that were added being exercised in an integration test.

Reply:
The `get` method of singleton manager. The singleton manager could only be accessed in the thread where it's created. (the server main thread for integration test).

But these tests will access singleton manger in the test thread which is not allowed.

Reply:
I have removed this macro because I created another PR to resolve this problem. See #34766